### ISSUE TITLE
Use XForwarded Middleware for Kubernetes  Deployments

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/fmtm#1650]
**Is your feature request related to a problem? Please describe.**

Described better than I could here:

https://safir.lsst.io/user-guide/x-forwarded.html

**Describe the solution you'd like**

Use a middleware to correctly pass the user IP through any proxy to the application server for logging. 

There is one provided here https://github.com/lsst-sqre/safir/tree/main/src/safir/middleware

The code is MIT, so to save pulling in the stuff we don't need we could just take the middleware and attribute the user.

**Comment by spwoodcock** [comment]
Hey Ronald! Thanks so much for the message and offer to contribute 🙏 

We have since integrated [Bunkerweb](https://github.com/bunkerity/bunkerweb) into the software stack, so that needs to be taken into account.

In production we specify the container IP range:
https://github.com/hotosm/fmtm/blob/fd654fdc4567299624388527e6a0d8bac5c0fa2d/deploy/compose.main.yaml#L38-L44

Then for the Bunkerweb (Nginx) proxy we whitelist the internal IPs:
https://github.com/hotosm/fmtm/blob/fd654fdc4567299624388527e6a0d8bac5c0fa2d/deploy/compose.main.yaml#L64

I wrote this a while ago & I'm not 100% certain what the cause is and if this is the possible solution - I think it is, but definitely worth double checking!
To avoid this becoming an [XY problem](https://xyproblem.info), I will describe what we need:
- Essentially, the user connects to a service via the the Nginx reverse proxy, which logs the IP of the internal service that the user connected to, instead of the IP of the connecting user.
- I believe the IP of the user needs to be forwarded, perhaps by XForwarded middleware, so that it can be displayed in the proxy logs.

An example of the nginx logs (I redacted some info):

```
# This one is from the ODK Collect mobile app
[NGINX.ACCESS] odk.dev.fmtm.hotosm.org 10.20.30.1 - - [08/Apr/2025:21:59:44 +0000] "GET /v1/key/xxxxxx/projects/14/forms/buildings/manifest HTTP/2.0" 304 0 "-" "org.odk.collect.android/v2024.3.5-entity-select-5-gd9dcd5e43e-dirty Dalvik/2.1.0 (Linux; U; Android 13; xxx Build/xxx)"

# This one is from FastAPI backend --> ODK Central
[NGINX.ACCESS] odk.dev.fmtm.hotosm.org 10.20.30.1 - - [08/Apr/2025:21:59:04 +0000] "POST /v1/sessions HTTP/1.1" 200 228 "-" "Python/3.12 aiohttp/3.11.2"

# Also FastAPI backend --> ODK Central
[NGINX.ACCESS] odk.dev.fmtm.hotosm.org 10.20.30.1 - - [08/Apr/2025:21:59:04 +0000] "GET /v1/projects/2/datasets/features.svc/Entities?$select=__id,+__system/updatedAt,+osm_id,+status,+task_id,+submission_ids&$filter=__system/updatedAt+gt+2025-04-08T20:59:03.075654Z HTTP/1.1" 200 146 "-" "Python/3.12 aiohttp/3.11.2"

# For the sync service
[NGINX.ACCESS] sync.dev.fmtm.hotosm.org 10.20.30.1 - - [08/Apr/2025:22:13:11 +0000] "GET /v1/shape?offset=-1&table=odk_entities&where=project_id%3D6 HTTP/2.0" 499 0 "https://mapper.dev.fmtm.hotosm.org/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:137.0) Gecko/xxx Firefox/xxx.0"
```

Let me know if I can help with any more info / debugging!

**Comment by ro-nald** [comment]
Thank you @spwoodcock! I'll take this on and keep this Issue updated. I'll let you know if I need any further guidance - my guess is that it will likely be around initial setup and testing after I'd had a look into Bunkerweb.

**Comment by ro-nald** [comment]
Thank you for assigning @spwoodcock!

And thank you for the reference to the XY problem! In terms of the 'Y', I've had a look at the [documentation to use the proposed Middleware](https://github.com/lsst-sqre/safir/blob/1c4991861708eb8f96aa447804a27daac3a4c4e2/docs/user-guide/x-forwarded.rst) and will look to add it after [the CORSMiddleware](https://github.com/hotosm/fmtm/blob/518d520010e41e2aa8fc0bc0a2859b290dd650c7/src/backend/app/main.py#L136C5-L143C6) once I've done a bit more research into whether this solves the problem.

To confirm this, or at least investigate 'X' further, I need to setup my local environment to be able to reproduce the nginx logs you provided. Just wanted to check, is a local instance the recommended way to test out this change for FMTM? As per https://docs.fmtm.dev/dev/Backend/

**Comment by spwoodcock** [comment]
Yep running locally would be the easiest way to debug this 😄 (hopefully there are no issues I'm unaware of forwarding IPs on the local setup)

Let me know if you have an issues!
(and feel free to correct the docs if you find anything isn't explained well)

**Comment by spwoodcock** [comment]
How did it go?

Any feedback if you had issues would be appreciated 😁

**Comment by spwoodcock** [comment]
Not necessarily needed now, as mappers don't use FieldTM, only managers - lower traffic and tracking is less important.
